src/wp-includes/blocks.php

```diff
@@ -819,11 +819,12 @@ function insert_hooked_blocks( &$parsed_anchor_block, $relative_position, $hooke
 	 *
 	 * @since 6.4.0
 	 *
-	 * @param string[]                $hooked_block_types The list of hooked block types.
-	 * @param string                  $relative_position  The relative position of the hooked blocks.
-	 *                                                    Can be one of 'before', 'after', 'first_child', or 'last_child'.
-	 * @param string                  $anchor_block_type  The anchor block type.
-	 * @param WP_Block_Template|array $context            The block template, template part, or pattern that the anchor block belongs to.
+	 * @param string[]                        $hooked_block_types The list of hooked block types.
+	 * @param string                          $relative_position  The relative position of the hooked blocks.
+	 *                                                            Can be one of 'before', 'after', 'first_child', or 'last_child'.
+	 * @param string                          $anchor_block_type  The anchor block type.
+	 * @param WP_Block_Template|WP_Post|array $context            The block template, template part, `wp_navigation` post type,
```

Comment:
This needs to be noted in the dev note as potentially it could break things if someone assumed:

```php
if ( ! $context instanceof WP_Block_Template ) // process as it was an array
```

Reply:
Good spot!